## Example Workflow for PDE Estimation

This provides an example workflow for a user defined equation. In a situation where we wish to analyze real data, the parameters would be different. However, here we consider an equation we analyzed in the accompanying paper.

In [1]:
import train

#### Parameter and Equation Defintions

Here we write out the number of points to sample, the number of epochs, and other related parameters for the algorithm. 

We also include the equation we wish to analyze, in this case, we're looking at the Hamilton Jacobi Bellman equation. 

In [2]:
params = {'n_epochs':30001,
          'n_points':10000,
          'n_grad_points':10000,
          'lambda_u':10,
          'lambda_f':1,
          'svd_init':500,
          'lambda_norm':1e-3,
          'width':50,
          'layers':4,
          'lr':0.002,
          'noise_levels': [1],
          'n_repetitions':1}

hjb = {'eqn_type':'hjb',
        'fcn':'x**2/(1+4*t)+1/2*log(1+4*t)',
        'domain':{'x':[-2,2],'t':[0,2]},
        'dictionary':'(u_tt,u_xx,u_t,u_x,u,u**2,u*u_x,u_x**2)',
        'err_vec':[0,-1,1,0,0,0,0,1]}


#### Running the algorithm

Here we simply run the train function and see how the error in the estimation of the law changes per epoch. We defined all our parameters in the above dictionaries.

In [3]:
stats = train.train(**hjb,**params)

-------------
% noise: 0.31622776601683794
width 50
depth 4
total number of parameters: 10703

###### Epoch 0 ######
Error in law (svd) 6.577063e-01
[0.0006, -0.0001, -0.0000, -0.0625, 0.0001, -0.0002, 0.1732, -0.9829]
Error in law (svd2) 9.379663e-01
[0.0183, -0.0227, 0.0014, -0.3339, 0.0010, -0.0028, 0.9240, 0.1841]
Eigenvalues [38.5658, 0.0262, 0.0196, 0.0046, 0.0013, 0.0001, 0.0000, 0.0000]
Learning rate: 0.002
Learning rate: 0.02
Loss u 8.658211146676196
Loss f 0.0
Error in law 0.932143176058335

###### Epoch 1 ######
Error in law (svd) 6.654994e-01
[0.0015, -0.0020, 0.0003, -0.1288, 0.0000, -0.0000, 0.2184, -0.9673]
Error in law (svd2) 9.224884e-01
[-0.0044, 0.0040, -0.0005, 0.4913, -0.0002, 0.0004, -0.8333, -0.2536]
Eigenvalues [68.2958, 0.0425, 0.0178, 0.0054, 0.0016, 0.0003, 0.0000, 0.0000]
Learning rate: 0.002
Learning rate: 0.02
Loss u 6.813535209346794
Loss f 0.0
Error in law 0.932143176058335

###### Epoch 100 ######
Error in law (svd) 7.461109e-01
[-0.0563, 0.1930, 0.0118


###### Epoch 800 ######
Error in law (svd) 2.433758e-01
[-0.1009, 0.5599, -0.7426, -0.0060, 0.1003, -0.0886, -0.0006, -0.3269]
Error in law (svd2) 8.727470e-01
[0.0668, 0.2780, 0.1796, 0.0509, -0.7119, 0.5262, -0.0308, -0.3144]
Eigenvalues [432.5950, 158.2224, 153.3585, 63.6939, 32.2740, 26.9281, 13.8680, 3.4589]
Learning rate: 0.0017046012261582844
Learning rate: 0.017046012261582845
Loss u 1.319279825943493
Loss f 0.0012126297650530708
Error in law 0.23153682916244742
Error in law (vec) 2.315368e-01
[-0.1000, 0.5678, -0.7381, -0.0022, 0.0793, -0.0735, -0.0024, -0.3333]
Loss f 0.0012126297650530708
Loss norm 0.0018966330225060133

###### Epoch 801 ######
Error in law (svd) 2.435078e-01
[-0.1009, 0.5593, -0.7429, -0.0060, 0.1007, -0.0887, -0.0006, -0.3271]
Error in law (svd2) 8.724889e-01
[0.0666, 0.2786, 0.1794, 0.0508, -0.7120, 0.5258, -0.0307, -0.3143]
Eigenvalues [433.0773, 158.3919, 153.5345, 63.7165, 32.2813, 26.9309, 13.8661, 3.4595]
Learning rate: 0.001704260305913053
Learning


###### Epoch 1500 ######
Error in law (svd) 2.186166e-01
[0.0737, -0.5497, 0.7324, 0.0053, -0.1109, 0.0941, -0.0026, 0.3672]
Error in law (svd2) 8.710545e-01
[0.0663, 0.2719, 0.1944, 0.0618, -0.6987, 0.5239, -0.0395, -0.3403]
Eigenvalues [510.0425, 160.3297, 157.0753, 61.8510, 30.4449, 27.3444, 13.7706, 3.3190]
Learning rate: 0.001481888364682653
Learning rate: 0.014818883646826531
Loss u 1.3169418193684703
Loss f 0.001114221967914852
Error in law 0.2066289963438352
Error in law (vec) 2.066290e-01
[-0.0728, 0.5566, -0.7283, -0.0006, 0.0929, -0.0810, -0.0001, -0.3732]
Loss f 0.001114221967914852
Loss norm 0.0019055835952330735

###### Epoch 1501 ######
Error in law (svd) 2.188439e-01
[0.0737, -0.5490, 0.7328, 0.0053, -0.1113, 0.0942, -0.0027, 0.3673]
Error in law (svd2) 8.707975e-01
[0.0661, 0.2726, 0.1941, 0.0617, -0.6989, 0.5235, -0.0394, -0.3402]
Eigenvalues [510.4938, 160.5403, 157.2752, 61.8771, 30.4689, 27.3499, 13.7692, 3.3201]
Learning rate: 0.0014815919870097166
Learning rate:


###### Epoch 2200 ######
Error in law (svd) 1.448209e-01
[-0.0395, 0.5486, -0.6917, -0.0010, 0.0843, -0.0674, 0.0002, -0.4555]
Error in law (svd2) 9.098825e-01
[-0.0655, -0.2364, -0.2582, -0.0796, 0.6968, -0.5254, 0.0527, 0.3200]
Eigenvalues [587.8080, 165.4458, 156.2721, 61.1192, 30.1762, 27.1580, 13.8033, 2.8973]
Learning rate: 0.0012882738154137137
Learning rate: 0.012882738154137137
Loss u 1.3155611487638947
Loss f 0.0008495042082943282
Error in law 0.13292444707406648
Error in law (vec) 1.329244e-01
[-0.0388, 0.5543, -0.6868, -0.0005, 0.0676, -0.0552, -0.0002, -0.4603]
Loss f 0.0008495042082943282
Loss norm 0.0018638142930603163

###### Epoch 2201 ######
Error in law (svd) 1.446460e-01
[-0.0394, 0.5486, -0.6915, -0.0010, 0.0842, -0.0673, 0.0002, -0.4556]
Error in law (svd2) 9.099934e-01
[-0.0655, -0.2363, -0.2584, -0.0796, 0.6968, -0.5254, 0.0527, 0.3199]
Eigenvalues [587.9597, 165.4380, 156.2617, 61.1166, 30.1753, 27.1585, 13.8038, 2.8966]
Learning rate: 0.001288016160650631
Lea


###### Epoch 2900 ######
Error in law (svd) 4.615646e-02
[-0.0102, 0.5603, -0.6222, 0.0061, 0.0238, -0.0147, -0.0057, -0.5458]
Error in law (svd2) 9.744955e-01
[-0.0537, -0.1971, -0.3436, -0.0919, 0.7060, -0.5245, 0.0594, 0.2337]
Eigenvalues [694.2107, 170.4120, 156.8639, 58.8187, 31.4310, 27.0074, 14.1909, 2.0642]
Learning rate: 0.0011199557692971168
Learning rate: 0.011199557692971169
Loss u 1.314605042429765
Loss f 0.0004691415095250171
Error in law 0.03501669524990367
Error in law (vec) 3.501670e-02
[-0.0089, 0.5657, -0.6150, -0.0004, 0.0051, -0.0008, -0.0026, -0.5493]
Loss f 0.0004691415095250171
Loss norm 0.0017477785129697305

###### Epoch 2901 ######
Error in law (svd) 4.584048e-02
[-0.0101, 0.5608, -0.6219, 0.0062, 0.0236, -0.0146, -0.0058, -0.5457]
Error in law (svd2) 9.747171e-01
[-0.0536, -0.1968, -0.3438, -0.0920, 0.7060, -0.5247, 0.0595, 0.2334]
Eigenvalues [693.8850, 170.3346, 156.7807, 58.8144, 31.4297, 27.0016, 14.1973, 2.0606]
Learning rate: 0.0011197317781432574
Lea


###### Epoch 3600 ######
Error in law (svd) 2.469127e-02
[-0.0057, 0.5673, -0.6029, 0.0075, 0.0074, -0.0016, -0.0065, -0.5607]
Error in law (svd2) 9.903864e-01
[-0.0401, -0.1967, -0.3449, -0.0928, 0.7191, -0.5278, 0.0594, 0.1814]
Eigenvalues [752.5886, 172.4224, 157.0593, 57.8505, 34.0302, 27.1098, 14.5809, 1.4216]
Learning rate: 0.000973629138599773
Learning rate: 0.00973629138599773
Loss u 1.3144543080531312
Loss f 0.00020881831824139414
Error in law 0.02308386774791949
Error in law (vec) 2.308387e-02
[-0.0059, 0.5678, -0.6026, 0.0001, 0.0046, 0.0001, -0.0027, -0.5608]
Loss f 0.00020881831824139414
Loss norm 0.0017446231102400506

###### Epoch 3601 ######
Error in law (svd) 2.549204e-02
[-0.0058, 0.5658, -0.6040, 0.0072, 0.0080, -0.0016, -0.0063, -0.5611]
Error in law (svd2) 9.898288e-01
[-0.0402, -0.1982, -0.3451, -0.0927, 0.7193, -0.5267, 0.0592, 0.1820]
Eigenvalues [754.6106, 173.0132, 157.5900, 57.8798, 34.0665, 27.1428, 14.5697, 1.4252]
Learning rate: 0.000973434412772053
Learn


###### Epoch 4300 ######
Error in law (svd) 2.176342e-02
[-0.0054, 0.5681, -0.6000, 0.0075, 0.0040, 0.0014, -0.0063, -0.5631]
Error in law (svd2) 9.935267e-01
[-0.0311, -0.2011, -0.3316, -0.0901, 0.7295, -0.5308, 0.0574, 0.1529]
Eigenvalues [791.6993, 174.1561, 157.6372, 57.4330, 36.3853, 27.2336, 14.8296, 1.2234]
Learning rate: 0.0008464206583135606
Learning rate: 0.008464206583135606
Loss u 1.314261525226842
Loss f 0.00016338170437711928
Error in law 0.02044582770522589
Error in law (vec) 2.044583e-02
[-0.0052, 0.5693, -0.5997, 0.0009, 0.0019, 0.0026, -0.0028, -0.5623]
Loss f 0.00016338170437711928
Loss norm 0.0017447605551089015

###### Epoch 4301 ######
Error in law (svd) 2.166468e-02
[-0.0054, 0.5684, -0.5999, 0.0075, 0.0039, 0.0014, -0.0063, -0.5630]
Error in law (svd2) 9.936044e-01
[-0.0311, -0.2009, -0.3316, -0.0902, 0.7295, -0.5309, 0.0574, 0.1528]
Eigenvalues [791.3254, 174.1027, 157.5794, 57.4344, 36.3880, 27.2306, 14.8329, 1.2228]
Learning rate: 0.0008462513741818979
Learn


###### Epoch 5000 ######
Error in law (svd) 2.161054e-02
[-0.0056, 0.5685, -0.5998, 0.0074, 0.0037, 0.0015, -0.0062, -0.5629]
Error in law (svd2) 9.940003e-01
[-0.0249, -0.2046, -0.3186, -0.0870, 0.7367, -0.5334, 0.0552, 0.1348]
Eigenvalues [820.5351, 175.2161, 157.8316, 57.1334, 38.3449, 27.3191, 15.0033, 1.1773]
Learning rate: 0.0007358324668161574
Learning rate: 0.007358324668161574
Loss u 1.314174026760178
Loss f 0.00014355151169353282
Error in law 0.020287591186612684
Error in law (vec) 2.028759e-02
[-0.0057, 0.5696, -0.5994, 0.0010, 0.0014, 0.0027, -0.0029, -0.5623]
Loss f 0.00014355151169353282
Loss norm 0.0017449422635797598

###### Epoch 5001 ######
Error in law (svd) 2.157730e-02
[-0.0056, 0.5686, -0.5998, 0.0075, 0.0036, 0.0016, -0.0062, -0.5628]
Error in law (svd2) 9.940440e-01
[-0.0249, -0.2044, -0.3186, -0.0870, 0.7367, -0.5335, 0.0552, 0.1347]
Eigenvalues [820.4984, 175.1914, 157.8166, 57.1349, 38.3475, 27.3194, 15.0049, 1.1774]
Learning rate: 0.0007356853003227942
Lear


###### Epoch 5700 ######
Error in law (svd) 2.243245e-02
[-0.0060, 0.5685, -0.6007, 0.0073, 0.0047, 0.0007, -0.0061, -0.5620]
Error in law (svd2) 9.933970e-01
[-0.0204, -0.2076, -0.3072, -0.0842, 0.7419, -0.5355, 0.0533, 0.1224]
Eigenvalues [845.1329, 176.0887, 157.9608, 56.9337, 40.0360, 27.3877, 15.1248, 1.1865]
Learning rate: 0.0006396930579406639
Learning rate: 0.006396930579406639
Loss u 1.3141069613398288
Loss f 0.00014708888698601328
Error in law 0.02105823840191398
Error in law (vec) 2.105824e-02
[-0.0061, 0.5695, -0.6002, -0.0004, 0.0024, 0.0018, -0.0022, -0.5616]
Loss f 0.00014708888698601328
Loss norm 0.0017441891727282785

###### Epoch 5701 ######
Error in law (svd) 2.244629e-02
[-0.0060, 0.5684, -0.6007, 0.0073, 0.0047, 0.0007, -0.0060, -0.5621]
Error in law (svd2) 9.933828e-01
[-0.0204, -0.2076, -0.3072, -0.0842, 0.7419, -0.5355, 0.0533, 0.1224]
Eigenvalues [845.2916, 176.1095, 157.9789, 56.9338, 40.0393, 27.3893, 15.1246, 1.1865]
Learning rate: 0.0006395651193290757
Lea


###### Epoch 6400 ######
Error in law (svd) 2.359796e-02
[-0.0064, 0.5682, -0.6018, 0.0074, 0.0059, -0.0002, -0.0060, -0.5611]
Error in law (svd2) 9.925844e-01
[-0.0171, -0.2099, -0.2978, -0.0815, 0.7459, -0.5370, 0.0516, 0.1134]
Eigenvalues [867.0713, 176.9076, 158.0975, 56.8033, 41.5369, 27.4430, 15.2123, 1.2135]
Learning rate: 0.0005561146413504409
Learning rate: 0.005561146413504409
Loss u 1.3140530101147738
Loss f 0.00015242990510062424
Error in law 0.022187026376644655
Error in law (vec) 2.218703e-02
[-0.0065, 0.5693, -0.6013, 0.0005, 0.0036, 0.0008, -0.0026, -0.5606]
Loss f 0.00015242990510062424
Loss norm 0.001745252092067723

###### Epoch 6401 ######
Error in law (svd) 2.354178e-02
[-0.0064, 0.5683, -0.6017, 0.0074, 0.0059, -0.0002, -0.0061, -0.5610]
Error in law (svd2) 9.926277e-01
[-0.0171, -0.2098, -0.2977, -0.0815, 0.7459, -0.5372, 0.0516, 0.1134]
Eigenvalues [866.7912, 176.8460, 158.0349, 56.8006, 41.5345, 27.4388, 15.2129, 1.2136]
Learning rate: 0.0005560034184221707
Le


###### Epoch 7100 ######
Error in law (svd) 2.451640e-02
[-0.0067, 0.5680, -0.6027, 0.0074, 0.0067, -0.0009, -0.0060, -0.5603]
Error in law (svd2) 9.920399e-01
[-0.0145, -0.2113, -0.2903, -0.0793, 0.7489, -0.5384, 0.0500, 0.1065]
Eigenvalues [886.5054, 177.5796, 158.1705, 56.7190, 42.8463, 27.4912, 15.2805, 1.2451]
Learning rate: 0.0004834560739488529
Learning rate: 0.0048345607394885285
Loss u 1.314006690685037
Loss f 0.00016018369141365928
Error in law 0.023165070647356026
Error in law (vec) 2.316507e-02
[-0.0068, 0.5691, -0.6023, 0.0006, 0.0044, 0.0002, -0.0026, -0.5597]
Loss f 0.00016018369141365928
Loss norm 0.001745714495754079

###### Epoch 7101 ######
Error in law (svd) 2.449481e-02
[-0.0067, 0.5681, -0.6027, 0.0074, 0.0067, -0.0009, -0.0060, -0.5603]
Error in law (svd2) 9.920587e-01
[-0.0145, -0.2112, -0.2903, -0.0793, 0.7489, -0.5384, 0.0500, 0.1064]
Eigenvalues [886.3724, 177.5571, 158.1454, 56.7189, 42.8471, 27.4896, 15.2811, 1.2453]
Learning rate: 0.0004833593827340632
Le


###### Epoch 7800 ######
Error in law (svd) 2.505228e-02
[-0.0069, 0.5680, -0.6032, 0.0074, 0.0069, -0.0011, -0.0060, -0.5597]
Error in law (svd2) 9.919046e-01
[-0.0126, -0.2119, -0.2847, -0.0773, 0.7513, -0.5395, 0.0487, 0.1007]
Eigenvalues [903.6103, 178.1565, 158.2479, 56.6648, 43.9809, 27.5383, 15.3370, 1.2754]
Learning rate: 0.0004202906344462737
Learning rate: 0.004202906344462737
Loss u 1.3139674373170762
Loss f 0.00016770865548470498
Error in law 0.023468693317448407
Error in law (vec) 2.346869e-02
[-0.0070, 0.5692, -0.6026, 0.0010, 0.0039, 0.0004, -0.0027, -0.5593]
Loss f 0.00016770865548470498
Loss norm 0.001746159073839047

###### Epoch 7801 ######
Error in law (svd) 2.504444e-02
[-0.0069, 0.5680, -0.6032, 0.0074, 0.0069, -0.0011, -0.0060, -0.5597]
Error in law (svd2) 9.919182e-01
[-0.0126, -0.2119, -0.2847, -0.0773, 0.7512, -0.5395, 0.0487, 0.1007]
Eigenvalues [903.7111, 178.1562, 158.2533, 56.6637, 43.9820, 27.5390, 15.3370, 1.2755]
Learning rate: 0.0004202065763193844
Le


###### Epoch 8500 ######
Error in law (svd) 2.539623e-02
[-0.0071, 0.5680, -0.6036, 0.0073, 0.0070, -0.0012, -0.0058, -0.5593]
Error in law (svd2) 9.918465e-01
[-0.0110, -0.2122, -0.2800, -0.0757, 0.7532, -0.5404, 0.0475, 0.0960]
Eigenvalues [918.7222, 178.6729, 158.3233, 56.6324, 44.9713, 27.5830, 15.3837, 1.3015]
Learning rate: 0.00036537800830678833
Learning rate: 0.0036537800830678834
Loss u 1.3139338048217652
Loss f 0.0001750317205600163
Error in law 0.02401533082576201
Error in law (vec) 2.401533e-02
[-0.0072, 0.5690, -0.6032, 0.0005, 0.0044, 0.0003, -0.0021, -0.5589]
Loss f 0.0001750317205600163
Loss norm 0.0017455120340111542

###### Epoch 8501 ######
Error in law (svd) 2.539331e-02
[-0.0071, 0.5680, -0.6036, 0.0073, 0.0070, -0.0012, -0.0058, -0.5593]
Error in law (svd2) 9.918505e-01
[-0.0110, -0.2122, -0.2800, -0.0757, 0.7532, -0.5404, 0.0475, 0.0960]
Eigenvalues [918.7744, 178.6713, 158.3249, 56.6339, 44.9736, 27.5841, 15.3844, 1.3017]
Learning rate: 0.000365304932705127
Lea


###### Epoch 9200 ######
Error in law (svd) 2.558991e-02
[-0.0072, 0.5681, -0.6038, 0.0071, 0.0071, -0.0012, -0.0056, -0.5590]
Error in law (svd2) 9.918546e-01
[-0.0097, -0.2122, -0.2762, -0.0741, 0.7547, -0.5412, 0.0464, 0.0920]
Eigenvalues [932.0688, 179.0854, 158.3762, 56.6251, 45.8393, 27.6303, 15.4213, 1.3245]
Learning rate: 0.000317639932971909
Learning rate: 0.00317639932971909
Loss u 1.3139053727577708
Loss f 0.00018055386161601342
Error in law 0.024319071404298807
Error in law (vec) 2.431907e-02
[-0.0073, 0.5691, -0.6034, 0.0003, 0.0046, 0.0001, -0.0022, -0.5585]
Loss f 0.00018055386161601342
Loss norm 0.0017455869429554914

###### Epoch 9201 ######
Error in law (svd) 2.566299e-02
[-0.0072, 0.5679, -0.6040, 0.0071, 0.0071, -0.0012, -0.0056, -0.5591]
Error in law (svd2) 9.917941e-01
[-0.0097, -0.2125, -0.2762, -0.0741, 0.7548, -0.5410, 0.0464, 0.0921]
Eigenvalues [932.5262, 179.1734, 158.4561, 56.6286, 45.8470, 27.6356, 15.4209, 1.3244]
Learning rate: 0.0003175764049853146
Lea


###### Epoch 9900 ######
Error in law (svd) 2.572124e-02
[-0.0073, 0.5683, -0.6040, 0.0069, 0.0069, -0.0012, -0.0054, -0.5586]
Error in law (svd2) 9.919255e-01
[-0.0087, -0.2121, -0.2731, -0.0727, 0.7560, -0.5419, 0.0454, 0.0888]
Eigenvalues [944.9109, 179.4803, 158.4415, 56.6452, 46.6477, 27.6843, 15.4528, 1.3479]
Learning rate: 0.00027613902513170044
Learning rate: 0.002761390251317004
Loss u 1.3138802840479724
Loss f 0.00018654880719826862
Error in law 0.024345602507256186
Error in law (vec) 2.434560e-02
[-0.0074, 0.5692, -0.6035, 0.0004, 0.0041, 0.0003, -0.0022, -0.5584]
Loss f 0.00018654880719826862
Loss norm 0.0017455461682285675

###### Epoch 9901 ######
Error in law (svd) 2.581357e-02
[-0.0073, 0.5679, -0.6042, 0.0068, 0.0070, -0.0011, -0.0054, -0.5588]
Error in law (svd2) 9.918569e-01
[-0.0087, -0.2124, -0.2732, -0.0726, 0.7560, -0.5416, 0.0454, 0.0888]
Eigenvalues [945.6910, 179.6086, 158.5639, 56.6502, 46.6581, 27.6926, 15.4523, 1.3477]
Learning rate: 0.000276083797326674
L


###### Epoch 10600 ######
Error in law (svd) 2.595917e-02
[-0.0075, 0.5682, -0.6043, 0.0067, 0.0068, -0.0010, -0.0052, -0.5584]
Error in law (svd2) 9.919891e-01
[0.0078, 0.2122, 0.2706, 0.0712, -0.7570, 0.5423, -0.0444, -0.0861]
Eigenvalues [958.2165, 179.9527, 158.5684, 56.6914, 47.4537, 27.7409, 15.4758, 1.3736]
Learning rate: 0.00024006037429629288
Learning rate: 0.002400603742962929
Loss u 1.3138473875301497
Loss f 0.00019369763842347302
Error in law 0.024651447383222506
Error in law (vec) 2.465145e-02
[-0.0076, 0.5692, -0.6038, 0.0004, 0.0042, 0.0002, -0.0016, -0.5580]
Loss f 0.00019369763842347302
Loss norm 0.0017450286383111247

###### Epoch 10601 ######
Error in law (svd) 2.601542e-02
[-0.0075, 0.5680, -0.6045, 0.0066, 0.0069, -0.0010, -0.0051, -0.5585]
Error in law (svd2) 9.919365e-01
[0.0078, 0.2124, 0.2707, 0.0712, -0.7571, 0.5422, -0.0444, -0.0861]
Eigenvalues [958.6598, 180.0226, 158.6319, 56.6933, 47.4587, 27.7449, 15.4749, 1.3734]
Learning rate: 0.00024001236222143363
L


###### Epoch 11300 ######
Error in law (svd) 2.636844e-02
[-0.0076, 0.5680, -0.6049, 0.0064, 0.0068, -0.0010, -0.0049, -0.5580]
Error in law (svd2) 9.919010e-01
[0.0071, 0.2124, 0.2684, 0.0698, -0.7580, 0.5427, -0.0434, -0.0840]
Eigenvalues [972.9460, 180.4941, 158.7104, 56.7810, 48.2902, 27.7969, 15.4949, 1.4005]
Learning rate: 0.0002086955412397467
Learning rate: 0.002086955412397467
Loss u 1.3138184750806723
Loss f 0.0002009794108791799
Error in law 0.02497499169299827
Error in law (vec) 2.497499e-02
[-0.0077, 0.5693, -0.6041, 0.0002, 0.0040, 0.0005, -0.0014, -0.5576]
Loss f 0.0002009794108791799
Loss norm 0.0017446849229628813

###### Epoch 11301 ######
Error in law (svd) 2.628548e-02
[-0.0076, 0.5683, -0.6047, 0.0065, 0.0068, -0.0010, -0.0049, -0.5578]
Error in law (svd2) 9.919802e-01
[0.0071, 0.2121, 0.2683, 0.0698, -0.7579, 0.5428, -0.0434, -0.0839]
Eigenvalues [972.4638, 180.4073, 158.6321, 56.7787, 48.2868, 27.7926, 15.4965, 1.4009]
Learning rate: 0.00020865380213149875
Learn


###### Epoch 12000 ######
Error in law (svd) 2.681009e-02
[-0.0078, 0.5681, -0.6053, 0.0063, 0.0068, -0.0011, -0.0046, -0.5574]
Error in law (svd2) 9.918573e-01
[0.0065, 0.2123, 0.2662, 0.0684, -0.7588, 0.5432, -0.0424, -0.0820]
Eigenvalues [988.3604, 180.9715, 158.7417, 56.9080, 49.1629, 27.8398, 15.5121, 1.4276]
Learning rate: 0.00018142864711022573
Learning rate: 0.0018142864711022573
Loss u 1.313787940332343
Loss f 0.00020825882770037982
Error in law 0.025668995198442065
Error in law (vec) 2.566900e-02
[-0.0079, 0.5691, -0.6048, 0.0002, 0.0044, 0.0001, -0.0013, -0.5570]
Loss f 0.00020825882770037982
Loss norm 0.0017448713715414411

###### Epoch 12001 ######
Error in law (svd) 2.682947e-02
[-0.0078, 0.5681, -0.6054, 0.0063, 0.0068, -0.0010, -0.0046, -0.5574]
Error in law (svd2) 9.918371e-01
[0.0065, 0.2124, 0.2662, 0.0685, -0.7588, 0.5431, -0.0424, -0.0820]
Eigenvalues [988.4813, 180.9928, 158.7566, 56.9083, 49.1647, 27.8403, 15.5115, 1.4276]
Learning rate: 0.00018139236138080367
L


###### Epoch 12700 ######
Error in law (svd) 2.725294e-02
[-0.0080, 0.5681, -0.6058, 0.0062, 0.0066, -0.0009, -0.0043, -0.5568]
Error in law (svd2) 9.919156e-01
[0.0059, 0.2122, 0.2644, 0.0672, -0.7595, 0.5435, -0.0415, -0.0801]
Eigenvalues [1005.1855, 181.5438, 158.8030, 57.0896, 50.0518, 27.8778, 15.5252, 1.4546]
Learning rate: 0.00015772428005269619
Learning rate: 0.0015772428005269617
Loss u 1.3137581255907915
Loss f 0.0002159458777836957
Error in law 0.02614050237439165
Error in law (vec) 2.614050e-02
[-0.0081, 0.5692, -0.6053, 0.0001, 0.0042, 0.0003, -0.0011, -0.5564]
Loss f 0.0002159458777836957
Loss norm 0.001744597685596932

###### Epoch 12701 ######
Error in law (svd) 2.724973e-02
[-0.0080, 0.5681, -0.6058, 0.0062, 0.0066, -0.0009, -0.0043, -0.5568]
Error in law (svd2) 9.919205e-01
[0.0059, 0.2122, 0.2644, 0.0672, -0.7596, 0.5435, -0.0415, -0.0801]
Eigenvalues [1005.2627, 181.5469, 158.8061, 57.0910, 50.0541, 27.8786, 15.5256, 1.4547]
Learning rate: 0.00015769273519668564
Le


###### Epoch 13400 ######
Error in law (svd) 2.762905e-02
[-0.0082, 0.5682, -0.6063, 0.0062, 0.0062, -0.0006, -0.0040, -0.5562]
Error in law (svd2) 9.921468e-01
[0.0054, 0.2119, 0.2630, 0.0660, -0.7602, 0.5439, -0.0406, -0.0781]
Eigenvalues [1023.2888, 182.1514, 158.8465, 57.3425, 50.9422, 27.9099, 15.5373, 1.4821]
Learning rate: 0.00013711698187898362
Learning rate: 0.001371169818789836
Loss u 1.3137277467180166
Loss f 0.00022418527726505755
Error in law 0.02655360641240957
Error in law (vec) 2.655361e-02
[-0.0082, 0.5693, -0.6057, 0.0000, 0.0037, 0.0006, -0.0006, -0.5559]
Loss f 0.00022418527726505755
Loss norm 0.00174412672442058

###### Epoch 13401 ######
Error in law (svd) 2.763384e-02
[-0.0082, 0.5682, -0.6063, 0.0062, 0.0062, -0.0006, -0.0040, -0.5562]
Error in law (svd2) 9.921417e-01
[0.0054, 0.2119, 0.2630, 0.0660, -0.7602, 0.5439, -0.0406, -0.0781]
Eigenvalues [1023.2766, 182.1535, 158.8474, 57.3423, 50.9428, 27.9096, 15.5371, 1.4821]
Learning rate: 0.00013708955848260783
Le


###### Epoch 14100 ######
Error in law (svd) 2.792495e-02
[-0.0084, 0.5684, -0.6066, 0.0062, 0.0055, -0.0001, -0.0036, -0.5557]
Error in law (svd2) 9.925096e-01
[0.0049, 0.2114, 0.2618, 0.0649, -0.7608, 0.5443, -0.0398, -0.0762]
Eigenvalues [1042.3918, 182.7625, 158.8684, 57.6823, 51.7967, 27.9366, 15.5475, 1.5100]
Learning rate: 0.00011920210834577928
Learning rate: 0.0011920210834577927
Loss u 1.3136981493017328
Loss f 0.0002323971226339153
Error in law 0.026948618572841546
Error in law (vec) 2.694862e-02
[-0.0084, 0.5694, -0.6060, 0.0003, 0.0031, 0.0012, -0.0004, -0.5554]
Loss f 0.0002323971226339153
Loss norm 0.0017441147916768575

###### Epoch 14101 ######
Error in law (svd) 2.797196e-02
[-0.0084, 0.5683, -0.6067, 0.0062, 0.0056, -0.0000, -0.0036, -0.5557]
Error in law (svd2) 9.924621e-01
[0.0049, 0.2116, 0.2618, 0.0650, -0.7609, 0.5442, -0.0398, -0.0762]
Eigenvalues [1042.6537, 182.8268, 158.9184, 57.6876, 51.8004, 27.9390, 15.5468, 1.5099]
Learning rate: 0.00011917826792411012



###### Epoch 14800 ######
Error in law (svd) 2.820219e-02
[-0.0085, 0.5685, -0.6070, 0.0063, 0.0048, 0.0006, -0.0034, -0.5552]
Error in law (svd2) 9.929419e-01
[0.0045, 0.2110, 0.2609, 0.0641, -0.7614, 0.5445, -0.0391, -0.0743]
Eigenvalues [1062.5070, 183.4644, 158.9113, 58.1340, 52.5880, 27.9567, 15.5548, 1.5374]
Learning rate: 0.00010362788357330948
Learning rate: 0.0010362788357330948
Loss u 1.3136672724737046
Loss f 0.0002406315334465251
Error in law 0.02724504406156898
Error in law (vec) 2.724504e-02
[-0.0085, 0.5695, -0.6063, 0.0002, 0.0023, 0.0018, -0.0003, -0.5549]
Loss f 0.0002406315334465251
Loss norm 0.0017439909379643452

###### Epoch 14801 ######
Error in law (svd) 2.819574e-02
[-0.0085, 0.5685, -0.6070, 0.0063, 0.0048, 0.0006, -0.0034, -0.5552]
Error in law (svd2) 9.929550e-01
[0.0045, 0.2110, 0.2609, 0.0640, -0.7614, 0.5445, -0.0391, -0.0743]
Eigenvalues [1062.6771, 183.4680, 158.9219, 58.1371, 52.5914, 27.9588, 15.5558, 1.5376]
Learning rate: 0.00010360715799659481
Lea


###### Epoch 15500 ######
Error in law (svd) 2.838855e-02
[-0.0086, 0.5686, -0.6072, 0.0064, 0.0039, 0.0013, -0.0032, -0.5549]
Error in law (svd2) 9.934795e-01
[0.0041, 0.2104, 0.2602, 0.0633, -0.7619, 0.5448, -0.0385, -0.0723]
Eigenvalues [1083.5509, 184.1645, 158.9377, 58.7163, 53.2787, 27.9738, 15.5629, 1.5625]
Learning rate: 9.008849258548893e-05
Learning rate: 0.0009008849258548893
Loss u 1.3136389941670346
Loss f 0.0002486721345094563
Error in law 0.027456081576533804
Error in law (vec) 2.745608e-02
[-0.0087, 0.5697, -0.6065, 0.0002, 0.0014, 0.0026, 0.0000, -0.5546]
Loss f 0.0002486721345094563
Loss norm 0.0017435725008108345

###### Epoch 15501 ######
Error in law (svd) 2.838872e-02
[-0.0086, 0.5686, -0.6072, 0.0064, 0.0039, 0.0013, -0.0032, -0.5549]
Error in law (svd2) 9.934792e-01
[0.0041, 0.2104, 0.2602, 0.0633, -0.7619, 0.5448, -0.0385, -0.0723]
Eigenvalues [1083.5150, 184.1658, 158.9363, 58.7166, 53.2798, 27.9735, 15.5629, 1.5626]
Learning rate: 9.007047488697182e-05
Learn


###### Epoch 16200 ######
Error in law (svd) 2.854514e-02
[-0.0087, 0.5688, -0.6073, 0.0066, 0.0030, 0.0021, -0.0030, -0.5546]
Error in law (svd2) 9.940633e-01
[0.0038, 0.2097, 0.2597, 0.0626, -0.7624, 0.5450, -0.0379, -0.0705]
Eigenvalues [1105.0860, 184.8796, 158.9413, 59.4316, 53.8371, 27.9839, 15.5698, 1.5842]
Learning rate: 7.831807633689863e-05
Learning rate: 0.0007831807633689863
Loss u 1.3136131179781114
Loss f 0.00025514124310775074
Error in law 0.02763498170457962
Error in law (vec) 2.763498e-02
[-0.0088, 0.5699, -0.6065, 0.0006, 0.0004, 0.0034, -0.0000, -0.5543]
Loss f 0.00025514124310775074
Loss norm 0.0017439682156432893

###### Epoch 16201 ######
Error in law (svd) 2.854693e-02
[-0.0087, 0.5688, -0.6073, 0.0066, 0.0030, 0.0021, -0.0030, -0.5546]
Error in law (svd2) 9.940636e-01
[0.0038, 0.2097, 0.2597, 0.0626, -0.7624, 0.5450, -0.0379, -0.0705]
Eigenvalues [1105.0923, 184.8822, 158.9414, 59.4327, 53.8383, 27.9838, 15.5699, 1.5842]
Learning rate: 7.830241272163124e-05
Lea


###### Epoch 16900 ######
Error in law (svd) 2.902945e-02
[-0.0088, 0.5680, -0.6080, 0.0066, 0.0027, 0.0026, -0.0028, -0.5547]
Error in law (svd2) 9.941552e-01
[0.0035, 0.2103, 0.2591, 0.0620, -0.7629, 0.5446, -0.0373, -0.0691]
Eigenvalues [1130.2186, 185.9533, 159.2347, 60.3413, 54.2637, 28.0113, 15.5726, 1.6009]
Learning rate: 6.80855113131316e-05
Learning rate: 0.000680855113131316
Loss u 1.3136903046939548
Loss f 0.0002607280700322166
Error in law 0.0279770893294265
Error in law (vec) 2.797709e-02
[-0.0089, 0.5699, -0.6068, 0.0011, 0.0001, 0.0038, -0.0001, -0.5540]
Loss f 0.0002607280700322166
Loss norm 0.0017445965642909997

###### Epoch 16901 ######
Error in law (svd) 2.858218e-02
[-0.0088, 0.5697, -0.6070, 0.0070, 0.0022, 0.0025, -0.0030, -0.5539]
Error in law (svd2) 9.946335e-01
[0.0035, 0.2084, 0.2587, 0.0619, -0.7627, 0.5459, -0.0374, -0.0688]
Eigenvalues [1125.3141, 185.2715, 158.6331, 60.2282, 54.2721, 27.9717, 15.5775, 1.6034]
Learning rate: 6.807189421086899e-05
Learning


###### Epoch 17600 ######
Error in law (svd) 2.898390e-02
[-0.0089, 0.5688, -0.6077, 0.0070, 0.0020, 0.0030, -0.0028, -0.5541]
Error in law (svd2) 9.947168e-01
[0.0033, 0.2090, 0.2584, 0.0614, -0.7632, 0.5454, -0.0369, -0.0676]
Eigenvalues [1150.3935, 186.3377, 158.9283, 61.2118, 54.5754, 27.9970, 15.5789, 1.6170]
Learning rate: 5.918987119690716e-05
Learning rate: 0.0005918987119690717
Loss u 1.3135683444517323
Loss f 0.00026528733839551876
Error in law 0.028228766437419735
Error in law (vec) 2.822877e-02
[-0.0089, 0.5696, -0.6072, 0.0013, 0.0003, 0.0036, 0.0000, -0.5539]
Loss f 0.00026528733839551876
Loss norm 0.0017448438527837393

###### Epoch 17601 ######
Error in law (svd) 2.898190e-02
[-0.0089, 0.5688, -0.6077, 0.0070, 0.0020, 0.0030, -0.0028, -0.5541]
Error in law (svd2) 9.947126e-01
[0.0033, 0.2090, 0.2583, 0.0614, -0.7632, 0.5454, -0.0369, -0.0676]
Eigenvalues [1150.2264, 186.3413, 158.9263, 61.2109, 54.5765, 27.9963, 15.5788, 1.6170]
Learning rate: 5.917803322266778e-05
Lea


###### Epoch 18300 ######
Error in law (svd) 2.910360e-02
[-0.0090, 0.5690, -0.6077, 0.0071, 0.0015, 0.0035, -0.0027, -0.5539]
Error in law (svd2) 9.950593e-01
[0.0031, 0.2086, 0.2579, 0.0609, -0.7635, 0.5455, -0.0365, -0.0664]
Eigenvalues [1172.9951, 187.0706, 158.9268, 62.1946, 54.8167, 28.0041, 15.5872, 1.6303]
Learning rate: 5.14564814853752e-05
Learning rate: 0.000514564814853752
Loss u 1.3135486637120302
Loss f 0.0002691779561685249
Error in law 0.028500969463168872
Error in law (vec) 2.850097e-02
[-0.0090, 0.5696, -0.6074, 0.0017, 0.0003, 0.0038, 0.0001, -0.5536]
Loss f 0.0002691779561685249
Loss norm 0.0017453808485625815

###### Epoch 18301 ######
Error in law (svd) 2.911186e-02
[-0.0090, 0.5689, -0.6077, 0.0071, 0.0015, 0.0035, -0.0027, -0.5539]
Error in law (svd2) 9.950393e-01
[0.0031, 0.2086, 0.2579, 0.0609, -0.7635, 0.5455, -0.0365, -0.0664]
Eigenvalues [1172.8209, 187.0663, 158.9210, 62.1923, 54.8118, 28.0022, 15.5857, 1.6301]
Learning rate: 5.1446190189078125e-05
Learni


###### Epoch 19000 ######
Error in law (svd) 2.920302e-02
[-0.0090, 0.5690, -0.6078, 0.0072, 0.0010, 0.0039, -0.0026, -0.5538]
Error in law (svd2) 9.953336e-01
[0.0029, 0.2083, 0.2576, 0.0604, -0.7638, 0.5456, -0.0361, -0.0655]
Eigenvalues [1195.2547, 187.7815, 158.9093, 63.1897, 54.9819, 28.0050, 15.5906, 1.6411]
Learning rate: 4.473348958720346e-05
Learning rate: 0.0004473348958720346
Loss u 1.3135307452373297
Loss f 0.00027244528129318277
Error in law 0.02866854847837673
Error in law (vec) 2.866855e-02
[-0.0090, 0.5697, -0.6075, 0.0019, 0.0001, 0.0041, 0.0000, -0.5534]
Loss f 0.00027244528129318277
Loss norm 0.001745691857870121

###### Epoch 19001 ######
Error in law (svd) 2.919314e-02
[-0.0090, 0.5690, -0.6078, 0.0072, 0.0010, 0.0039, -0.0026, -0.5538]
Error in law (svd2) 9.953485e-01
[0.0029, 0.2082, 0.2576, 0.0604, -0.7638, 0.5456, -0.0361, -0.0655]
Eigenvalues [1195.2830, 187.7787, 158.9069, 63.1911, 54.9850, 28.0055, 15.5914, 1.6412]
Learning rate: 4.472454288928602e-05
Learn


###### Epoch 19700 ######
Error in law (svd) 2.927172e-02
[-0.0090, 0.5691, -0.6078, 0.0073, 0.0006, 0.0042, -0.0025, -0.5537]
Error in law (svd2) 9.956058e-01
[0.0028, 0.2079, 0.2574, 0.0599, -0.7640, 0.5457, -0.0357, -0.0646]
Eigenvalues [1217.8250, 188.4858, 158.8936, 64.1962, 55.1126, 28.0074, 15.5960, 1.6506]
Learning rate: 3.8888883049984336e-05
Learning rate: 0.00038888883049984337
Loss u 1.3135139055725287
Loss f 0.00027549158422693045
Error in law 0.028798168060934787
Error in law (vec) 2.879817e-02
[-0.0090, 0.5696, -0.6077, 0.0021, 0.0000, 0.0040, 0.0000, -0.5534]
Loss f 0.00027549158422693045
Loss norm 0.0017458389762318712

###### Epoch 19701 ######
Error in law (svd) 2.927397e-02
[-0.0090, 0.5691, -0.6078, 0.0073, 0.0006, 0.0042, -0.0025, -0.5537]
Error in law (svd2) 9.956021e-01
[0.0028, 0.2080, 0.2574, 0.0599, -0.7640, 0.5457, -0.0357, -0.0646]
Eigenvalues [1217.9066, 188.4933, 158.8989, 64.1987, 55.1126, 28.0078, 15.5960, 1.6506]
Learning rate: 3.8881105273374344e-05



###### Epoch 20400 ######
Error in law (svd) 2.932995e-02
[-0.0090, 0.5691, -0.6078, 0.0074, 0.0003, 0.0045, -0.0025, -0.5536]
Error in law (svd2) 9.958214e-01
[0.0027, 0.2077, 0.2572, 0.0594, -0.7642, 0.5458, -0.0353, -0.0640]
Eigenvalues [1240.5341, 189.1939, 158.8824, 65.1976, 55.2146, 28.0094, 15.6006, 1.6586]
Learning rate: 3.3807897367970666e-05
Learning rate: 0.00033807897367970664
Loss u 1.3134980162587122
Loss f 0.00027794529877893387
Error in law 0.02893147844951451
Error in law (vec) 2.893148e-02
[-0.0090, 0.5695, -0.6078, 0.0024, 0.0001, 0.0040, 0.0000, -0.5532]
Loss f 0.00027794529877893387
Loss norm 0.0017461621977159967

###### Epoch 20401 ######
Error in law (svd) 2.933365e-02
[-0.0090, 0.5691, -0.6078, 0.0074, 0.0003, 0.0045, -0.0025, -0.5536]
Error in law (svd2) 9.958107e-01
[0.0027, 0.2077, 0.2572, 0.0594, -0.7642, 0.5458, -0.0353, -0.0640]
Eigenvalues [1240.4266, 189.1992, 158.8833, 65.1977, 55.2139, 28.0088, 15.6002, 1.6586]
Learning rate: 3.3801135788497075e-05
L


###### Epoch 21100 ######
Error in law (svd) 2.936062e-02
[-0.0090, 0.5693, -0.6078, 0.0075, -0.0001, 0.0047, -0.0024, -0.5535]
Error in law (svd2) 9.960509e-01
[0.0026, 0.2074, 0.2571, 0.0589, -0.7643, 0.5458, -0.0350, -0.0634]
Eigenvalues [1262.9359, 189.8873, 158.8658, 66.1865, 55.3022, 28.0113, 15.6068, 1.6658]
Learning rate: 2.9390762469936724e-05
Learning rate: 0.0002939076246993672
Loss u 1.313482825348827
Loss f 0.00028019416271532087
Error in law 0.029020717797216527
Error in law (vec) 2.902072e-02
[-0.0091, 0.5696, -0.6079, 0.0025, 0.0000, 0.0041, -0.0001, -0.5531]
Loss f 0.00028019416271532087
Loss norm 0.0017464085455818003

###### Epoch 21101 ######
Error in law (svd) 2.936482e-02
[-0.0090, 0.5692, -0.6078, 0.0074, -0.0000, 0.0047, -0.0024, -0.5535]
Error in law (svd2) 9.960413e-01
[0.0026, 0.2074, 0.2571, 0.0589, -0.7643, 0.5458, -0.0350, -0.0634]
Eigenvalues [1262.9758, 189.8883, 158.8628, 66.1875, 55.2998, 28.0104, 15.6061, 1.6658]
Learning rate: 2.9384884317442738e-05


###### Epoch 21800 ######
Error in law (svd) 2.944873e-02
[-0.0090, 0.5691, -0.6079, 0.0075, -0.0003, 0.0050, -0.0023, -0.5536]
Error in law (svd2) 9.961792e-01
[0.0026, 0.2074, 0.2571, 0.0584, -0.7645, 0.5457, -0.0346, -0.0629]
Eigenvalues [1286.7876, 190.6849, 158.9284, 67.1945, 55.3697, 28.0178, 15.6109, 1.6720]
Learning rate: 2.555074363727258e-05
Learning rate: 0.0002555074363727258
Loss u 1.3134758428910063
Loss f 0.0002821266883961726
Error in law 0.02909871890066309
Error in law (vec) 2.909872e-02
[-0.0091, 0.5696, -0.6079, 0.0027, 0.0000, 0.0042, 0.0000, -0.5531]
Loss f 0.0002821266883961726
Loss norm 0.0017465389725485013

###### Epoch 21801 ######
Error in law (svd) 2.933148e-02
[-0.0090, 0.5696, -0.6076, 0.0076, -0.0004, 0.0050, -0.0024, -0.5534]
Error in law (svd2) 9.963106e-01
[0.0026, 0.2069, 0.2570, 0.0584, -0.7644, 0.5461, -0.0347, -0.0628]
Eigenvalues [1284.9078, 190.4761, 158.7525, 67.1485, 55.3701, 28.0050, 15.6113, 1.6728]
Learning rate: 2.5545633488545126e-05
Lea


###### Epoch 22500 ######
Error in law (svd) 2.941811e-02
[-0.0090, 0.5694, -0.6077, 0.0075, -0.0006, 0.0052, -0.0023, -0.5534]
Error in law (svd2) 9.964168e-01
[0.0025, 0.2070, 0.2571, 0.0580, -0.7646, 0.5459, -0.0343, -0.0625]
Eigenvalues [1308.9392, 191.2802, 158.8203, 68.1467, 55.4262, 28.0114, 15.6152, 1.6787]
Learning rate: 2.2212438383842675e-05
Learning rate: 0.00022212438383842673
Loss u 1.3134533890984534
Loss f 0.0002842350295470294
Error in law 0.02916118059817669
Error in law (vec) 2.916118e-02
[-0.0091, 0.5696, -0.6080, 0.0028, 0.0001, 0.0042, 0.0000, -0.5530]
Loss f 0.0002842350295470294
Loss norm 0.0017467169734680854

###### Epoch 22501 ######
Error in law (svd) 2.941994e-02
[-0.0090, 0.5694, -0.6077, 0.0075, -0.0006, 0.0052, -0.0023, -0.5534]
Error in law (svd2) 9.964114e-01
[0.0025, 0.2070, 0.2571, 0.0580, -0.7646, 0.5459, -0.0343, -0.0625]
Eigenvalues [1308.9562, 191.2808, 158.8179, 68.1475, 55.4249, 28.0108, 15.6148, 1.6786]
Learning rate: 2.2207995896165908e-05
L


###### Epoch 23200 ######
Error in law (svd) 2.944466e-02
[-0.0090, 0.5695, -0.6077, 0.0076, -0.0009, 0.0054, -0.0023, -0.5534]
Error in law (svd2) 9.965780e-01
[0.0025, 0.2068, 0.2571, 0.0576, -0.7647, 0.5460, -0.0340, -0.0621]
Eigenvalues [1332.5808, 191.9802, 158.7965, 69.1162, 55.4739, 28.0114, 15.6193, 1.6848]
Learning rate: 1.931029585519628e-05
Learning rate: 0.0001931029585519628
Loss u 1.3134387181581608
Loss f 0.0002862638753294501
Error in law 0.029237042525699773
Error in law (vec) 2.923704e-02
[-0.0090, 0.5696, -0.6081, 0.0029, 0.0000, 0.0043, 0.0000, -0.5529]
Loss f 0.0002862638753294501
Loss norm 0.0017467778710104779

###### Epoch 23201 ######
Error in law (svd) 2.944435e-02
[-0.0090, 0.5695, -0.6077, 0.0076, -0.0009, 0.0054, -0.0023, -0.5534]
Error in law (svd2) 9.965798e-01
[0.0025, 0.2068, 0.2571, 0.0576, -0.7647, 0.5460, -0.0340, -0.0621]
Eigenvalues [1332.5877, 191.9817, 158.7967, 69.1176, 55.4747, 28.0115, 15.6195, 1.6849]
Learning rate: 1.9306433796025242e-05
Le


###### Epoch 23900 ######
Error in law (svd) 2.945905e-02
[-0.0090, 0.5696, -0.6077, 0.0076, -0.0011, 0.0055, -0.0022, -0.5533]
Error in law (svd2) 9.967387e-01
[0.0025, 0.2066, 0.2571, 0.0572, -0.7647, 0.5460, -0.0338, -0.0618]
Eigenvalues [1356.6993, 192.6752, 158.7623, 70.0778, 55.5164, 28.0111, 15.6238, 1.6910]
Learning rate: 1.678732967410048e-05
Learning rate: 0.00016787329674100478
Loss u 1.3134239869143751
Loss f 0.00028821786377280167
Error in law 0.02931148182697034
Error in law (vec) 2.931148e-02
[-0.0090, 0.5696, -0.6081, 0.0030, -0.0000, 0.0043, 0.0000, -0.5528]
Loss f 0.00028821786377280167
Loss norm 0.0017469924624941061

###### Epoch 23901 ######
Error in law (svd) 2.947539e-02
[-0.0090, 0.5695, -0.6077, 0.0076, -0.0011, 0.0056, -0.0022, -0.5533]
Error in law (svd2) 9.967201e-01
[0.0025, 0.2067, 0.2571, 0.0572, -0.7647, 0.5460, -0.0338, -0.0618]
Eigenvalues [1356.8942, 192.7008, 158.7845, 70.0840, 55.5160, 28.0125, 15.6237, 1.6908]
Learning rate: 1.678397220816566e-05



###### Epoch 24600 ######
Error in law (svd) 2.949764e-02
[-0.0090, 0.5696, -0.6077, 0.0077, -0.0013, 0.0057, -0.0022, -0.5533]
Error in law (svd2) 9.968695e-01
[0.0025, 0.2065, 0.2573, 0.0568, -0.7648, 0.5460, -0.0335, -0.0616]
Eigenvalues [1381.5026, 193.4080, 158.7538, 71.0354, 55.5519, 28.0121, 15.6281, 1.6967]
Learning rate: 1.4593998957871998e-05
Learning rate: 0.00014593998957871998
Loss u 1.3134090409816024
Loss f 0.0002902221706748406
Error in law 0.029344874418862178
Error in law (vec) 2.934487e-02
[-0.0090, 0.5696, -0.6082, 0.0030, 0.0000, 0.0043, 0.0000, -0.5528]
Loss f 0.0002902221706748406
Loss norm 0.001746945578557877

###### Epoch 24601 ######
Error in law (svd) 2.948724e-02
[-0.0090, 0.5696, -0.6077, 0.0077, -0.0014, 0.0057, -0.0022, -0.5532]
Error in law (svd2) 9.968829e-01
[0.0025, 0.2064, 0.2572, 0.0568, -0.7648, 0.5460, -0.0335, -0.0616]
Eigenvalues [1381.4532, 193.3908, 158.7395, 71.0334, 55.5521, 28.0113, 15.6283, 1.6968]
Learning rate: 1.4591080158080426e-05
L


###### Epoch 25300 ######
Error in law (svd) 2.954646e-02
[-0.0090, 0.5696, -0.6077, 0.0077, -0.0015, 0.0059, -0.0021, -0.5532]
Error in law (svd2) 9.969696e-01
[0.0025, 0.2065, 0.2574, 0.0565, -0.7648, 0.5459, -0.0333, -0.0614]
Eigenvalues [1407.0512, 194.1621, 158.7552, 71.9812, 55.5814, 28.0138, 15.6322, 1.7022]
Learning rate: 1.2687235535200234e-05
Learning rate: 0.00012687235535200234
Loss u 1.3133960974974244
Loss f 0.000292005914528371
Error in law 0.02940165397127361
Error in law (vec) 2.940165e-02
[-0.0090, 0.5696, -0.6082, 0.0032, 0.0000, 0.0043, 0.0000, -0.5527]
Loss f 0.000292005914528371
Loss norm 0.00174710718179234

###### Epoch 25301 ######
Error in law (svd) 2.948545e-02
[-0.0090, 0.5698, -0.6076, 0.0077, -0.0016, 0.0059, -0.0022, -0.5531]
Error in law (svd2) 9.970433e-01
[0.0025, 0.2062, 0.2573, 0.0565, -0.7648, 0.5461, -0.0333, -0.0614]
Eigenvalues [1405.8765, 194.0491, 158.6661, 71.9533, 55.5823, 28.0076, 15.6326, 1.7028]
Learning rate: 1.2684698088093195e-05
Learn


###### Epoch 26000 ######
Error in law (svd) 2.954091e-02
[-0.0090, 0.5698, -0.6076, 0.0077, -0.0018, 0.0060, -0.0021, -0.5531]
Error in law (svd2) 9.971262e-01
[0.0025, 0.2062, 0.2575, 0.0563, -0.7648, 0.5460, -0.0331, -0.0613]
Eigenvalues [1431.6609, 194.8117, 158.6773, 72.8791, 55.6086, 28.0100, 15.6370, 1.7081]
Learning rate: 1.1029598260922351e-05
Learning rate: 0.00011029598260922352
Loss u 1.3133783629924878
Loss f 0.0002939279295186895
Error in law 0.029451364829808774
Error in law (vec) 2.945136e-02
[-0.0090, 0.5697, -0.6082, 0.0033, -0.0000, 0.0042, 0.0000, -0.5526]
Loss f 0.0002939279295186895
Loss norm 0.001747193441075337

###### Epoch 26001 ######
Error in law (svd) 2.954146e-02
[-0.0090, 0.5698, -0.6076, 0.0077, -0.0018, 0.0060, -0.0021, -0.5531]
Error in law (svd2) 9.971257e-01
[0.0025, 0.2062, 0.2575, 0.0563, -0.7648, 0.5460, -0.0331, -0.0613]
Eigenvalues [1431.6764, 194.8127, 158.6794, 72.8799, 55.6085, 28.0102, 15.6370, 1.7080]
Learning rate: 1.1027392341270168e-05



###### Epoch 26700 ######
Error in law (svd) 2.956383e-02
[-0.0090, 0.5698, -0.6076, 0.0078, -0.0020, 0.0061, -0.0021, -0.5531]
Error in law (svd2) 9.972337e-01
[0.0025, 0.2060, 0.2577, 0.0560, -0.7648, 0.5460, -0.0329, -0.0613]
Eigenvalues [1456.8765, 195.5083, 158.6414, 73.7614, 55.6323, 28.0093, 15.6417, 1.7136]
Learning rate: 9.588537822902599e-06
Learning rate: 9.5885378229026e-05
Loss u 1.3133626191827406
Loss f 0.00029583158566455546
Error in law 0.02950064662165371
Error in law (vec) 2.950065e-02
[-0.0090, 0.5697, -0.6083, 0.0034, 0.0000, 0.0042, 0.0000, -0.5526]
Loss f 0.00029583158566455546
Loss norm 0.0017471925918904087

###### Epoch 26701 ######
Error in law (svd) 2.956380e-02
[-0.0090, 0.5698, -0.6076, 0.0078, -0.0020, 0.0061, -0.0021, -0.5531]
Error in law (svd2) 9.972349e-01
[0.0025, 0.2060, 0.2577, 0.0560, -0.7648, 0.5461, -0.0329, -0.0613]
Eigenvalues [1456.9237, 195.5090, 158.6413, 73.7628, 55.6325, 28.0093, 15.6418, 1.7137]
Learning rate: 9.586620115338019e-06
Lear


###### Epoch 27400 ######
Error in law (svd) 2.959003e-02
[-0.0090, 0.5699, -0.6076, 0.0078, -0.0021, 0.0062, -0.0020, -0.5530]
Error in law (svd2) 9.973170e-01
[0.0025, 0.2059, 0.2579, 0.0558, -0.7648, 0.5461, -0.0328, -0.0613]
Eigenvalues [1481.4836, 196.1831, 158.5979, 74.6024, 55.6512, 28.0072, 15.6458, 1.7190]
Learning rate: 8.335757604787431e-06
Learning rate: 8.335757604787431e-05
Loss u 1.31334664443408
Loss f 0.0002976926082428501
Error in law 0.029560116689066895
Error in law (vec) 2.956012e-02
[-0.0090, 0.5697, -0.6083, 0.0035, -0.0000, 0.0042, -0.0000, -0.5525]
Loss f 0.0002976926082428501
Loss norm 0.0017472441825266878

###### Epoch 27401 ######
Error in law (svd) 2.958813e-02
[-0.0090, 0.5699, -0.6076, 0.0078, -0.0021, 0.0062, -0.0020, -0.5530]
Error in law (svd2) 9.973220e-01
[0.0025, 0.2059, 0.2579, 0.0558, -0.7648, 0.5461, -0.0328, -0.0613]
Eigenvalues [1481.6461, 196.1862, 158.5995, 74.6055, 55.6521, 28.0077, 15.6461, 1.7191]
Learning rate: 8.334090453266474e-06
Lea


###### Epoch 28100 ######
Error in law (svd) 2.961440e-02
[-0.0089, 0.5700, -0.6076, 0.0078, -0.0023, 0.0063, -0.0020, -0.5529]
Error in law (svd2) 9.974018e-01
[0.0025, 0.2058, 0.2582, 0.0557, -0.7648, 0.5461, -0.0327, -0.0614]
Eigenvalues [1506.2063, 196.8515, 158.5533, 75.4163, 55.6696, 28.0057, 15.6507, 1.7247]
Learning rate: 7.2466580545580356e-06
Learning rate: 7.246658054558035e-05
Loss u 1.3133301721221682
Loss f 0.00029950794935710364
Error in law 0.029598665546992874
Error in law (vec) 2.959867e-02
[-0.0090, 0.5698, -0.6084, 0.0036, 0.0000, 0.0042, 0.0000, -0.5524]
Loss f 0.00029950794935710364
Loss norm 0.0017473840691350326

###### Epoch 28101 ######
Error in law (svd) 2.961563e-02
[-0.0089, 0.5700, -0.6076, 0.0078, -0.0023, 0.0063, -0.0020, -0.5529]
Error in law (svd2) 9.973987e-01
[0.0025, 0.2058, 0.2582, 0.0557, -0.7648, 0.5461, -0.0327, -0.0614]
Eigenvalues [1506.1882, 196.8516, 158.5532, 75.4166, 55.6690, 28.0055, 15.6505, 1.7247]
Learning rate: 7.245208722947125e-06



###### Epoch 28800 ######
Error in law (svd) 2.965927e-02
[-0.0089, 0.5700, -0.6076, 0.0078, -0.0023, 0.0064, -0.0019, -0.5529]
Error in law (svd2) 9.974408e-01
[0.0026, 0.2058, 0.2584, 0.0555, -0.7647, 0.5460, -0.0326, -0.0615]
Eigenvalues [1530.2843, 197.5280, 158.5267, 76.1951, 55.6830, 28.0043, 15.6549, 1.7301]
Learning rate: 6.299853648519091e-06
Learning rate: 6.29985364851909e-05
Loss u 1.3133141773251626
Loss f 0.0003013503353856645
Error in law 0.02964037705533803
Error in law (vec) 2.964038e-02
[-0.0090, 0.5698, -0.6084, 0.0038, 0.0000, 0.0042, -0.0000, -0.5524]
Loss f 0.0003013503353856645
Loss norm 0.0017474358788485032

###### Epoch 28801 ######
Error in law (svd) 2.963007e-02
[-0.0089, 0.5701, -0.6076, 0.0078, -0.0024, 0.0063, -0.0019, -0.5529]
Error in law (svd2) 9.974758e-01
[0.0026, 0.2057, 0.2584, 0.0555, -0.7647, 0.5461, -0.0326, -0.0615]
Eigenvalues [1529.7572, 197.4732, 158.4835, 76.1816, 55.6831, 28.0014, 15.6550, 1.7304]
Learning rate: 6.298593677789387e-06
Lear


###### Epoch 29500 ######
Error in law (svd) 2.966425e-02
[-0.0089, 0.5701, -0.6076, 0.0078, -0.0025, 0.0064, -0.0019, -0.5528]
Error in law (svd2) 9.975173e-01
[0.0026, 0.2056, 0.2587, 0.0555, -0.7647, 0.5461, -0.0326, -0.0617]
Eigenvalues [1552.9194, 198.1159, 158.4478, 76.9129, 55.6964, 28.0000, 15.6601, 1.7361]
Learning rate: 5.476752965844174e-06
Learning rate: 5.4767529658441746e-05
Loss u 1.3132962903064158
Loss f 0.0003033400921959977
Error in law 0.029693195575928498
Error in law (vec) 2.969320e-02
[-0.0089, 0.5698, -0.6084, 0.0039, 0.0000, 0.0042, 0.0000, -0.5523]
Loss f 0.0003033400921959977
Loss norm 0.0017475213661063839

###### Epoch 29501 ######
Error in law (svd) 2.966519e-02
[-0.0089, 0.5701, -0.6076, 0.0078, -0.0025, 0.0064, -0.0019, -0.5528]
Error in law (svd2) 9.975160e-01
[0.0026, 0.2056, 0.2587, 0.0555, -0.7647, 0.5461, -0.0326, -0.0617]
Eigenvalues [1552.9242, 198.1158, 158.4467, 76.9134, 55.6961, 27.9998, 15.6600, 1.7361]
Learning rate: 5.4756576152510055e-06
L

We see the final recovered vector is [-0.0089, 0.5699, -0.6084, 0.0039, 0.0000, 0.0042, 0.0000, -0.5522] with SNR of 0.5. The true vector is [0,-1,1,0,0,0,0,1] so the final error is ~3e-2.